### PR TITLE
test(x/ecocredit): migrate remaining core tests

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #893

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/core/features/msg_mint_batch_credits.feature [hunk 1]
```diff
@@ -1,67 +1,141 @@
...
 
+  Rule: The issuer must be the issuer of the credit batch
+
+    Background:
+      Given a credit type with abbreviation "C"
+      And a credit class with id "C01" and issuer alice
+      And a project with id "C01-001"
+      And a credit batch with denom "C01-001-20200101-20210101-001" and issuer alice
+
+    Scenario: the issuer is not the credit batch issuer
```
Comment (technicallyty) [inline x/ecocredit/server/core/features/msg_mint_batch_credits.feature:53]: same as the other review comment, not too sure on the wording here. it makes sense in the context of the message field, but if the reader doesn't have that context, could be confusing

**Review by technicallyty — APPROVED** [review]
lgtm, 1 nit wrt wording

**Review by aleem1314 — APPROVED** [review]
lgtm

File: x/ecocredit/server/core/features/msg_mint_batch_credits.feature (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/server/core/features/msg_mint_batch_credits.feature:53]: I agree. A bit awkward but leaving for now and tracking in the issue for writing guidelines: #1242

**Pull request merged**